app/models/vote.rb

```diff
@@ -0,0 +1,25 @@
+class Vote < ActiveRecord::Base
+  belongs_to :user
+  belongs_to :review
+
+  validates :user, presence: true
+  validates :review, presence: true
+  validates :user_id, uniqueness: {
+    scope: :review_id, message: "User can only vote once per review."
+  }
+
+  def self.user_already_voted(user, review)
+    Vote.where(user_id: user.id, review_id: review)
+  end
+
+  def self.same_vote(user, review, value)
+    review_id = review.id
+    Vote.where(user_id: user.id, review_id: review_id, value: value).empty?
```

Comment:
So this returns `true` if there is NOT a same vote already in existence? That seems backwards based on the name. I'd flip that so that this returns true if there IS a "same vote" already in existence.